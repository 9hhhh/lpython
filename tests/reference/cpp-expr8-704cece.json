{
    "basename": "cpp-expr8-704cece",
    "cmd": "lpython --no-color --show-cpp {infile}",
    "infile": "tests/expr8.py",
    "infile_hash": "c1844ccb51b77fb43b14fb6836c09ee3154ecfc8f672deee149b765a",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "cpp-expr8-704cece.stdout",
    "stdout_hash": "e5aafce83d8193fa58ffbfe5269b173eb9c454f0589b5c920c04da0e",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}