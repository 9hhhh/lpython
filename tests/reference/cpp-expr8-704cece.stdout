#include <iostream>
#include <string>
#include <vector>
#include <cassert>
#include <cmath>
#include <Kokkos_Core.hpp>
#include <lfortran_intrinsics.h>

template <typename T>
Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
{
    Kokkos::View<T*> r("r", v.size());
    for (size_t i=0; i < v.size(); i++) {
        r(i) = v[i];
    }
    return r;
}

void test_binop()
{
    int x;
    float x2;
    x = pow(2, 3);
    x = pow((float)(2), 3.500000);
    x = 54 - 100;
    x2 = 3.454000 - 765.430000 + 534.600000;
    x2 = 5346.565000*3.450000;
    x2 = pow(5346.565000, 3.450000);
    x = true + true;
    x = true - false;
    x = true*false;
    x = pow(false, false);
}

namespace {

void main2() {
}

}
int main(int argc, char* argv[])
{
    Kokkos::initialize(argc, argv);
    main2();
    Kokkos::finalize();
    return 0;
}
